src/EventListener/ExceptionListener.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\DieNicelyException;
+use Contao\CoreBundle\Exception\ResponseExceptionInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+
+/**
+ * Handle exceptions and create a proper response containing the error screen when debug mode is not active.
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ */
+class ExceptionListener
+{
+    /**
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * Create a new instance.
+     */
+    public function __construct($environment)
+    {
+        $this->environment = $environment;
+    }
+
+    /**
+     * Forwards the request to the Frontend class if there is a page object.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     */
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        // Search if an response is somewhere in the exception list.
+        $response = $this->checkResponseException($event->getException());
+        if ($response instanceof Response) {
+            $event->setResponse($response);
+
+            return;
+        }
+
+        // FIXME: log the exception here if we do not find a better place to log via monolog from outside.
+
+        // FIXME: do we really want this?
+        // If not in prod, exit here.
+        if ($this->environment !== 'prod') {
+            return;
+        }
+
+        $exception = $event->getException();
```

Comment:
This should go into a `checkDieNicelyException` just the same like `checkResponseException`.